pkg/cloudprovider/vsphereparavirtual/vmoperator/client/virtualmachine_client_test.go

```diff
@@ -0,0 +1,325 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	vmopv1alpha1 "github.com/vmware-tanzu/vm-operator-api/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgotesting "k8s.io/client-go/testing"
+
+	dynamicfake "k8s.io/client-go/dynamic/fake"
+)
+
+func initVMTest() (*virtualMachines, *dynamicfake.FakeDynamicClient) {
+	scheme := runtime.NewScheme()
+	_ = vmopv1alpha1.AddToScheme(scheme)
+	fc := dynamicfake.NewSimpleDynamicClient(scheme)
+	vms := newVirtualMachines(NewFakeClientSet(fc).V1alpha1(), "test-ns")
+	return vms, fc
+}
+
+func TestVMCreate(t *testing.T) {
+	vms, fc := initVMTest()
```

Comment:
Should this line be moved into the for loop?

If I see correctly the tests only work because PrependReactor happens in the last case which seems a bit brittle when we add more testcases

Reply:
Same below in other cases

And same in virtualmachineservices_client_test.go

Reply:
good catch! update it